### PR TITLE
Added build --destination and --source flags

### PR DESCRIPTION
**Pull request by tarebyte**
From issue #3417

### CODE REVIEW

File: lib/jekyll/command.rb [hunk 1]
```diff
@@ -49,6 +49,8 @@ def configuration_from_options(options)
       # Returns nothing
       def add_build_options(c)
         c.option 'config',  '--config CONFIG_FILE[,CONFIG_FILE2,...]', Array, 'Custom configuration file'
+        c.option 'destination', '--destination <destination>', 'The current folder will be generated into <destination>'
+        c.option 'source', '--source <source> --destination <destination>', 'The <source> folder will be generated into <destination>'
```
Comment (parkr) [inline lib/jekyll/command.rb]: ?

**Comment by parkr** [comment]
Please also add the short versions, `-s` and `-d`.

File: lib/jekyll/command.rb (hunk 1 above)
Comment (tarebyte) [inline lib/jekyll/command.rb]: Is this not the correct place for these flags? Or are the flags not correct?

File: lib/jekyll/command.rb [hunk 2]
```diff
@@ -49,6 +49,8 @@ def configuration_from_options(options)
       # Returns nothing
       def add_build_options(c)
         c.option 'config',  '--config CONFIG_FILE[,CONFIG_FILE2,...]', Array, 'Custom configuration file'
+        c.option 'destination', '-d', '--destination <destination>', 'The current folder will be generated into <destination>'
+        c.option 'source', '-s', '--source <source> --destination <destination>', 'The <source> folder will be generated into <destination>'
```
Comment (parkr) [inline lib/jekyll/command.rb]: You can only have one switch per line, you cannot mix them.

**Comment by parkr** [comment]
Please also use `SCREAMING_SNAKE_CASE` instead of `<brackets>` for the argument fillers.

**Comment by tarebyte** [comment]
@parkr should be fixed now, thanks for the feedback!

**Comment by tarebyte** [comment]
@parkr does this need anything else added to it? I think I caught everything you mentioned.

**Comment by parkr** [comment]
These are redundant because they're global: https://github.com/jekyll/jekyll/blob/master/bin/jekyll#L22-L26

They should be removed from being global if we want to do this.

**Pull request merged**